### PR TITLE
web: move `HierarchicalLocationsView` styles out from the `.theme-redesign`

### PR DESCRIPTION
**Pull request by valerybugakov**
## Changes

- Moved `HierarchicalLocationsView` styles out from the `.theme-redesign`.
- Converted `HierarchicalLocationsView` styles to CSS modules.
- Improved resizable handler styles provided by `HierarchicalLocationsView`.

Part of the https://github.com/sourcegraph/sourcegraph/issues/20847.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
Déjà vu 👀

File: client/branded/src/components/panel/views/HierarchicalLocationsView.module.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    &:hover {
+        opacity: 1;
+    }
+}
+
+.group-list {
+    flex: 1;
+    border-right: none;
+    overflow-x: hidden;
+}
```
Comment (pdubroy) [inline client/branded/src/components/panel/views/HierarchicalLocationsView.module.scss:29]: Why don't you need the `overflow-y: visible` here that was referred to by the comment?

**Review by pdubroy — APPROVED** [review]
Thanks for breaking it up. Looks like a nice simplication of a lot of the CSS! +1 to what @eseliger said about CSS modules making things a lot cleaner :-)

File: client/branded/src/components/panel/views/HierarchicalLocationsView.module.scss (hunk 1 above)
Comment (valerybugakov) [reply client/branded/src/components/panel/views/HierarchicalLocationsView.module.scss:29]: In the pre-redesign version, `&__list` also had `overflow-y: auto` which [was deleted](https://github.com/sourcegraph/sourcegraph/pull/22008/files#diff-a4ae857fea8a70e39230f43e429b6037cff31fbd1b379c0960a3f42e4cb97c0eL13) during the cleanup.
`overflow-y: visible` was required to override it.

**Pull request merged**